js/main.js

```diff
@@ -20,4 +20,77 @@ $entryForm.addEventListener('submit', function (event) {
   data.entries.unshift(submittedData);
   $image.setAttribute('src', 'images/placeholder-image-square.jpg');
   $entryForm.reset();
+
+  const newData = renderEntry(submittedData);
+  $unorderedList.prepend(newData);
+  toggleNoEntries();
+  viewSwap('entries');
 });
+
+function renderEntry(entry) {
+  const $newLi = document.createElement('li');
+  const $rowDiv = document.createElement('div');
+  const $columnHalfDiv = document.createElement('div');
+  const $imgElement = document.createElement('img');
+  const $secondColumnHalf = document.createElement('div');
+  const $titleElement = document.createElement('h3');
+  const $noteElement = document.createElement('p');
+
+  $rowDiv.setAttribute('class', 'row');
+  $columnHalfDiv.setAttribute('class', 'column-half');
+  $imgElement.setAttribute('class', 'image');
+  $imgElement.setAttribute('src', entry.url);
+  $secondColumnHalf.setAttribute('class', 'column-half');
+  $titleElement.textContent = entry.title;
+  $noteElement.textContent = entry.notes;
+
+  $newLi.appendChild($rowDiv);
+  $rowDiv.appendChild($columnHalfDiv);
+  $columnHalfDiv.appendChild($imgElement);
+
+  $rowDiv.appendChild($secondColumnHalf);
+  $secondColumnHalf.appendChild($titleElement);
+  $secondColumnHalf.appendChild($noteElement);
+
+  return $newLi;
+}
+
+const $unorderedList = document.querySelector('.no-bullets');
+
+document.addEventListener('DOMContentLoaded', function (event) {
+  for (let i = 0; i < data.entries.length; i++) {
+    $unorderedList.appendChild(renderEntry(data.entries[i]));
+  }
+  viewSwap(data.view);
+  toggleNoEntries();
+});
+
+function toggleNoEntries() {
+  const $noEntryText = document.querySelector('.no-entry');
+  if (data.entries.length !== 0) {
+    $noEntryText.className = 'no-entry hidden';
+  } else {
+    $noEntryText.className = 'no-entry';
+  }
+}
+
+function viewSwap(name) {
+  const $entriesView = document.querySelector('[data-view=entries]');
+  const $entryFormView = document.querySelector('[data-view=entry-form]');
+  if (name !== 'entries' && name !== 'entry-form') {
+    name = name.target.getAttribute('id');
+  }
```

Comment:
Not quite sure I understand the purpose of this? The only things that `name` will ever be is one of those two strings

Reply:
This is how I got my 'entries' anchor and 'new' button to switch between the views. Since I am passing in entries on the submit event listener, it will skip this if statement. But outside of that, this if statement will set the argument as the value of the id that I am clicking on.